src/NuGet.Services.Staging.Common/SecretReaderFactory.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Specialized;
+using Microsoft.Extensions.Configuration;
+using System.Security.Cryptography.X509Certificates;
+using NuGet.Services.KeyVault;
+
+namespace NuGet.Services.Staging.Common
+{
+    public class SecretReaderFactory : ISecretReaderFactory
+    {
+        private string _vaultName;
+        private string _clientId;
+        private string _certificateThumbprint;
+        private string _storeName;
+        private string _storeLocation;
+        private bool _validateCertificate;
+
+        public SecretReaderFactory(IConfigurationRoot config)
+        {
+            _vaultName = config[Constants.KeyVaultVaultNameKey];
+            _clientId = config[Constants.KeyVaultClientIdKey];
+            _certificateThumbprint = config[Constants.KeyVaultCertificateThumbprintKey];
+            _storeName = config[Constants.KeyVaultStoreNameKey];
+            _storeLocation = config[Constants.KeyVaultStoreLocationKey];
+            _validateCertificate = bool.Parse(config[Constants.KeyVaultValidateCertificateKey]);
+        }
+
+        public ISecretReader CreateSecretReader()
```

Comment:
>CreateSecretReader [](start = 29, length = 18)

since we need caching in Staging web app, lets make this guy get a parameter useCaching, that will determine the reader created.
Also, please add CachingReader NuGet.Services.KeyVault.. it's replicated in too many services.